pyiso/eu.py

```diff
@@ -0,0 +1,252 @@
+                              headers={'X-Ajax-call': 'true'})
+
+        msg = r.text
+        if msg == 'ok':
+            return True
+        elif msg == 'non_exists_user_or_bad_password':
+            # TODO throw error
+            return 'Wrong email or password'
+        elif msg == 'not_human':
+            return 'This account is not allowed to access web portal'
+        elif msg == 'suspended_use':
+            return 'User is suspended'
+        else:
+            return 'Unknown error:' + str(msg)
+
+    def fetch_entsoe(self, url, payload, count=0):
+        if not getattr(self, 'session', None):
+            self.auth()
+
+        r = self.session.get(url, params=payload)
+        # TODO error checking
+        if len(r.text) == 0:
+            if count > 3:  # try 3 times to get response
+                self.logger.warn('Request failed, no response found after %i attempts' % count)
+                return False
+            # throttled
+            sleep(5)
+            return self.fetch_entsoe(url, payload, count + 1)
+        if 'UNKNOWN_EXCEPTION' in r.text:
+            self.logger.warn('UNKNOWN EXCEPTION')
+            return False
+        return r.text
+
+    def construct_payload(self):
+        format_str = '%d.%m.%Y'
+        date_str = self.options['start_at'].strftime(format_str) + ' 00:00|UTC|DAY'
+        (TSO_ID, ) = [
+            i['ENTSOe_ID'] for i in control_areas if i['Code'] == self.options['control_area']]
+
+        payload = {
+            'name': '',
+            'defaultValue': 'false',
+            'viewType': 'TABLE',
+            'areaType': 'CTA',
+            'atch': 'false',
+            'dateTime.dateTime': date_str,
+            'biddingZone.values': TSO_ID,
+            'dateTime.timezone': 'UTC',
+            'dateTime.timezone_input': 'UTC',
+            'exportType': 'CSV',
+            'dataItem': 'ALL',
+            'timeRange': 'DEFAULT',
+        }
+        return payload
+
+    def parse_load_response(self, response):
+        df = pd.read_csv(StringIO(response))
+
+        # get START_TIME_UTC as tz-aware datetime
+        df['START_TIME_UTC'], df['END_TIME_UTC'] = zip(
+            *df['Time (UTC)'].apply(lambda x: x.split('-')))
+
+        # Why do these methods only work on Index and not Series?
+        df.set_index(df.START_TIME_UTC, inplace=True)
+        df.index = pd.to_datetime(df.index, utc=True, format='%d.%m.%Y %H:%M ')
+        df.START_TIME_UTC = df.index.to_pydatetime()
+
+        # find column name and choose which to return and which to drop
+        (forecast_load_col, ) = [c for c in df.columns if 'Day-ahead Total Load Forecast [MW]' in c]
+        (actual_load_col, ) = [c for c in df.columns if 'Actual Total Load [MW]' in c]
+        if self.options['forecast']:
+            load_col = forecast_load_col
+            drop_load_col = actual_load_col
+
+            # drop extra columns
+            df = df[self.options['start_at']:self.options['end_at']]
```

Comment:
This is enforced by test_load, is it the behavior we want outside of forecast=True?